src/DragMode.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * File containing drag based event functions.
+ * @author Dawn Moore
+ */
+
+import {Point} from "./AEG/Point";
+import {tree} from "./index";
+import {redrawCut} from "./index";
+
+const canvas: HTMLCanvasElement = <HTMLCanvasElement>document.getElementById("canvas");
+const res: CanvasRenderingContext2D | null = canvas.getContext("2d");
+if (res === null) {
+    throw Error("2d rendering context not supported");
+}
+const ctx: CanvasRenderingContext2D = res;
+
+let originPoint: Point;
+export let offSet: Point = new Point(0, 0); //No offset at the start
+let wasOut: boolean;
```

Comment:
comments on all global explaining what they are. We should be doing this in all mode files